chain/messagepool/messagepool.go

```diff
@@ -1213,6 +1214,9 @@ func (mp *MessagePool) remove(ctx context.Context, from address.Address, nonce u
 			return
 		}
 	}
+
+	// Record the current size of the Mpool
+	stats.Record(ctx, metrics.MpoolMessageCount.M(int64(mp.currentSize)))
```

Comment:
Any reason we're not updating this on add as well?

Reply:
I was considering it, but opted out because addLocked looks like it can be called fairly frequently in a very short amount of time, and I did not want to add "unnecessary" code there. The default metric scrape interval is 10s, so having it just on remove (HeadChange or pruneMessage) seemed adequate.

Happy to add it there as well if you think its okay/more correct?

Reply:
My concern is, e.g., if a node gets stuck (or slow) we won't get a picture of the actual number of messages and could draw the wrong conclusions.

I'm also not sure if doing it on every add is a huge issue. Especially given that we emit a journal event and and publish the change on a pubsub channel.

https://github.com/filecoin-project/lotus/blob/29fbfaf13df2fa35f84fd45dabd98b09324debfe/chain/messagepool/messagepool.go#L1012-L1022

Unless there's some hot path I'm missing here, I'd just go for accuracy.

Reply:
Agreed! Added it here: https://github.com/filecoin-project/lotus/pull/11361/commits/d121760172cbbcf9e0dd0f4ba963450a0ffc3bb1